app/src/main/java/edu/usf/sas/pal/muser/util/EventCaptureUtils.java

```diff
@@ -0,0 +1,43 @@
+package edu.usf.sas.pal.muser.util;
+
+import android.content.Context;
+
+import com.simplecity.amp_library.ShuttleApplication;
+import com.simplecity.amp_library.model.Song;
+
+import edu.usf.sas.pal.muser.firebase.model.Event;
+import edu.usf.sas.pal.muser.firebase.model.SongData;
+
+
+
+public class EventCaptureUtils {
+
+    /**
+     * Function to populate the Event data class.
+     * @param song - The song on which the event happened.
+     * @param capturedEvent - The event that was captured.
+     * @param context - The context of the Fragment.
+     * @return - Event object
+     */
+    public static Event captureEvent(Song song, String capturedEvent, Context context){
```

Comment:
I have the `context` as an argument because functions like `song.getBitrate()` need the context to be passed as an argument. Also, I'm passing the `event` as a String so I don't quite understand why I need to pass the enum itself. Here's the code that captures the event. 

                if(isPlaying())
                    event = EventProvider.EVENT_PLAY_MANUAL.getEvent();
                else
                    event = EventProvider.EVENT_PAUSE.getEvent();
                Log.d(TAG, "onViewCreated: " + event);
                EventCaptureUtils.captureEvent(song, event, getContext());

Reply:
>I have the context as an argument because functions like song.getBitrate() need the context to be passed as an argument.

Ok, sorry, I missed that

>I have the context as an argument because functions like song.getBitrate() need the context to be passed as an argument. Also, I'm passing the event as a String so I am not quite sure why I may have to pass the enum itself. Here's the code that captures the event.

Generally it's better to have variables strongly and specifically typed as much as possible to avoid accidentally passing some other `String` value later that's not actually a member of the enum. So the model class should have the enum type for the variable as well. We just then need to figure out how this gets persisted to Firestore as a String. Looks like Firebase may take care of this for us - https://stackoverflow.com/a/51565374/937715.

Reply:
You're right. It's better to have the enum passed and looks like Firebase does make work little easy. Thank you for the link.